### PR TITLE
Accept special characters in county names

### PR DESCRIPTION
**Pull request by SherinV**
Recognize county names with special chars in header parser. Test case added.
For issue https://github.com/deardurham/ciprs-reader/issues/43

### CODE REVIEW

File: ciprs_reader/parser/section/header.py [hunk 1]
```diff
@@ -13,7 +13,7 @@ class CaseDetails(HeaderParser):
     """Extract County and File No from header on top of first page"""
 
     pattern = (
-        r"\s*Case (Details|Summary) for Court Case[\s:]+(?P<county>(\w\s*)+) (?P<fileno>\w+)"
+        r"\s*Case (Details|Summary) for Court Case[\s:]+(?P<county>(\w\s*.*)+) (?P<fileno>\w+)"
```
Comment (robert-w-gries) [inline ciprs_reader/parser/section/header.py:16]: @copelco are we sure about using `.*` here? This regex looks good for the testcases we have and the generally expected output, but I'm curious if you've run into edge cases in the wild where unwanted characters would be matched.

If we definitely want to pull in all characters (including potential whitespace after the county), then this could simplify the regex a bit:

```
\s*Case (Details|Summary) for Court Case[\s:]+(?P<county>.+) (?P<fileno>\w+)
```

File: tests/parsers/test_header.py [hunk 2]
```diff
@@ -20,6 +20,10 @@
         {"county": "OLD HANOVER", "fileno": "99FN9999999"},
         " Case Summary for Court Case: OLD HANOVER 99FN9999999",
     ),
+    (
+        {"county": "GUILFORD-GR", "fileno": "15CR000000"},
+        " Case Summary for Court Case: GUILFORD-GR 15CR000000",
+    ),
```
Comment (robert-w-gries) [inline tests/parsers/test_header.py:26]: Thanks for adding the test case!

**Review by robert-w-gries — COMMENTED** [review]
Thanks for the contribution! Before merging, I'd like to see if there's any more edge cases we need to handle.

File: ciprs_reader/parser/section/header.py (hunk 1 above)
Comment (copelco) [reply ciprs_reader/parser/section/header.py:16]: @SherinV yes, thanks for your contribution!

@robert-w-gries Yes I think ``.*`` should be fine. I like your suggestion.

**Review by robert-w-gries — APPROVED** [review]

**Comment by robert-w-gries** [comment]
Thanks for the contribution!
